src/core/HatAllocator.ts

```diff
@@ -37,15 +39,27 @@ export class HatAllocator {
         this.toggleDecorations
       ),
 
-      vscode.window.onDidChangeTextEditorVisibleRanges(
-        this.addDecorationsDebounced
+      // An Event which fires when the active editor has changed. Note that the event also fires when the active editor changes to undefined.
+      vscode.window.onDidChangeActiveTextEditor(
+        this.addDecorationsDebouncedDefault
       ),
-      vscode.window.onDidChangeActiveTextEditor(this.addDecorationsDebounced),
-      vscode.window.onDidChangeVisibleTextEditors(this.addDecorationsDebounced),
+      // An Event which fires when the array of visible editors has changed.
+      vscode.window.onDidChangeVisibleTextEditors(
+        this.addDecorationsDebouncedDefault
+      ),
+      // An event that is emitted when a text document is changed. This usually happens when the contents changes but also when other things like the dirty-state changes.
+      vscode.workspace.onDidChangeTextDocument(
+        this.addDecorationsDebouncedDefault
+      ),
+      // An Event which fires when the selection in an editor has changed.
       vscode.window.onDidChangeTextEditorSelection(
-        this.addDecorationsDebounced
+        this.addDecorationsDebouncedDefault
       ),
-      vscode.workspace.onDidChangeTextDocument(this.addDecorationsDebounced)
+      // An Event which fires when the visible ranges of an editor has changed.
+      vscode.window.onDidChangeTextEditorVisibleRanges(() =>
+        // Increase debounced delay to stop the decorations moving when scrolling. 25Hz
+        this.addDecorationsDebounced(40)
```

Comment:
Move to constant at top?

Reply:
Idea is that if more events want to deviate from the default we can add a different delay for each single event. A constant at the top would make more sense if we actually reused it multiple times.

Reply:
As below, I have a philosophical objection but I think it's prob fine here as well